redaxo/src/core/assets/standard.js

```diff
@@ -799,3 +799,20 @@ var handleKeyEvents = function (event) {
         }
     }
 }
+
+function searchfield_init(selector_id = '') {
+
+    var $css_selector = selector_id;
+
+    $($css_selector).on('input propertychange', function () {
+        var $this = $(this);
+        var visible = Boolean($this.val());
+        $this.siblings('.form-control-clear').toggleClass('hidden', !visible);
+    }).trigger('propertychange');
+
+    $('.form-control-clear, .clear-button').click(function () {
+        event.stopPropagation();
+        $(this).siblings('input[type="text"]').val('').trigger("keyup")
+            .trigger('propertychange').focus();
+    });
+}
```

Comment:
ich gehe davon aus, dass es aktuell so gedacht ist, dass man das searchfield nur nutzen kann, wenn man zuvor diese methode aufgerufen hat, oder?

also der clear button etc funktioniert ja sonst nicht?
(falls wir das fragment so gestalten wollen, dass man diese methode nicht zwangsweise aufrufen muss, müssten wir die JS api ändern)

daher würd ich es so machen

```suggestion
/**
 * @param {string} selector_id
 */
function rex_searchfield_init(selector_id) {

    $(selector_id).on('input propertychange', function () {
        var $this = $(this);
        var visible = Boolean($this.val());
        $this.siblings('.form-control-clear').toggleClass('hidden', !visible);
    }).trigger('propertychange');

    $(selector_id).find('.form-control-clear, .clear-button').click(function () {
        event.stopPropagation();
        $(this).siblings('input[type="text"]').val('').trigger("keyup")
            .trigger('propertychange').focus();
    });
}
```

ungetestet.

Reply:
da das fragment unter `core/form/search.php` liegt, würde ich die methode ggf. auch `rex_form_search_init` nennen (oder das fragment noch verschieben, wenn es nicht nur für rex_form gedacht ist)

Reply:
Ich habe nochmal drüber nachgedacht. Und die Verwendungsstellen müssen ja immer noch auf das korrekte Unterelement ein keyUp legen: https://github.com/redaxo/redaxo/blob/586ef80eea5bd74a527187ffc4417b12d6aba6df/redaxo/src/core/pages/packages.list.php#L161-L174

Ich könnte mir vorstellen, dass wir das mit in die api holen, und man stattdessen an die init() ein callback übergibt, welches immer nur neue search-values erhält.

So in der Art:

```js
rex_searchfield_init('#my-search-field', function (search) {
    // search is current search value
});
```

Reply:
Ich habe die Änderung von @staabm übernommen. Mein Ziel war eigentlich ein allgemeines Feld anzubieten unabhängig wie die Suche selbst realisiert ist.